### PR TITLE
Landing Page Implementation

### PR DESCRIPTION
**Pull request by RosaGao**
Stored the meeting name in global store w/ redux toolkit; still working on styling and the Add button

### CODE REVIEW

**Review by K02D — DISMISSED** [review]

**Comment by RosaGao** [comment]
Kiron we may wanna try using redux toolkit. I just realized it is much less confusing than trying to figure out where to pass props around

**Comment by K02D** [comment]
Yeah I agree. I think we should Redux Toolkit anyway since Semesterly uses it, right?

**Comment by JJamesWWang** [comment]
> Yeah I agree. I think we should Redux Toolkit anyway since Semesterly uses it, right?

Yep

File: src/pages/Landing.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        justify="center"
+        style={{ minHeight: "100vh" }}
+      >
+  
+      <FormControl>
+        <TextField
+          fullWidth 
+          placeholder="Name the Meeting:"
+          aria-label="Name the Meeting:"
+          onChange={(event)=>dispatch(createMeeting(event.target.value))}
```
Comment (K02D) [inline src/pages/Landing.tsx]: I think here you dispatch a createMeeting action every time the text field changes: 
image

Maybe instead we should dispatch a createMeeting action only when the Create button is clicked?

Comment (JJamesWWang) [reply src/pages/Landing.tsx]: I would suggest for the meeting name on the landing page to be a React state (`useState("")`), and to dispatch a redux action when create is pressed (rename it to `setMeetingName()` maybe?)

**Review by RosaGao — COMMENTED** [review]
yeah this looks good. I'll take a closer look at the code tmr!

Kiron do you think we may get rid of Landing since all it does is to render createMeeting?

**Comment by K02D** [comment]
Yes I think we could. Should I go ahead and replace Landing with CreateMeeting?

**Comment by RosaGao** [comment]
No actually we can keep it for now. When I refresh the /create page, the meeting name disappears. I am not sure if it's because we haven't post it to the api endpoint so everything input.

**Comment by RosaGao** [comment]
Also I'll restyle the landing page. Grid is causing error message in the console

**Comment by K02D** [comment]
Okay sure. I think the meeting name disappears because we're not storing it anywhere. We'll have to use the endpoints as you said: a POST to create a meeting and a GET to fetch a meeting with a certain id

**Comment by RosaGao** [comment]
Yeah I'll try working on the API related functionalities. Also I am having trouble getting the landing page text field span the 100% width. Of course styling is of lower priority compared to API and generating id.

**Comment by RosaGao** [comment]
I guess our next main focus is to generate unique URLs for each meeting and get connected to the associated API endpoint.

**Comment by JJamesWWang** [comment]
> I guess our next main focus is to generate unique URLs for each meeting and get connected to the associated API endpoint.

The urls can be determined based solely on the id of the calendar in the following process

1. POST to create a new meeting
2. Response has an `id` field
3. Redirect to `/:id` on the frontend
4. You can then fetch based on that `id` (`api/calendars/:id`)

So you won't have to generate any URLs, you can directly use the id you get from the backend response

**Comment by RosaGao** [comment]
I see. Thanks so much for the clarification!

**Comment by RosaGao** [comment]
Screen Shot 2022-03-21 at 3 59 52 PM

Right now by clicking Create, the name and id should be saved

**Comment by yesterd3v** [comment]
If this is ready to merge, you can/should change it from draft to "ready for review", which can be found somewhere below.

**Review by yesterd3v — CHANGES_REQUESTED** [review]
Fix merge conflicts

**Review by yesterd3v — APPROVED** [review]
LGTM

**Pull request merged**